compiler/src/dotty/tools/dotc/transform/PostInlining.scala

```diff
@@ -0,0 +1,29 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import Contexts.*
+import DenotTransformers.IdentityDenotTransformer
+import Decorators.*
+import ast.tpd.*
+
+/** A phase that adds mirror support for anonymous mirrors created at inlining. */
+class PostInlining extends MacroTransform, IdentityDenotTransformer:
+  thisPhase =>
+
+  override def phaseName: String = PostInlining.name
+  override def changesMembers = true
+
+  override def run(using Context): Unit =
+    if ctx.compilationUnit.needsMirrorSupport then super.run
```

Comment:
```suggestion
    if ctx.compilationUnit.needsMirrorSupport && ctx.compilationUnit.needsInlining then super.run
```

Reply:
We don't need that. needsMirrorSupport is only set of something was inlined.

Reply:
We should document that in the definition of `needsMirrorSupport`. Future changes might easily break that assumption.